src/components/Jobs/JobDetails/JobDetails.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { Suspense, useCallback, useEffect, useRef, useState } from 'react'
+import ReactGA from 'react-ga4'
+import {
+    generatePath,
+    NavLink,
+    Redirect,
+    Route,
+    Switch,
+    useHistory,
+    useLocation,
+    useParams,
+    useRouteMatch,
+} from 'react-router-dom'
+import { URLS } from '../../../config'
+import AppConfig from '../../app/details/appConfig/AppConfig'
+import AppOverview from '../../app/details/appOverview/AppOverview'
+import CDDetails from '../../app/details/cdDetails/CDDetails'
+import CIDetails from '../../app/details/cIDetails/CIDetails'
+import TriggerView from '../../app/details/triggerView/TriggerView'
+import { getAppMetaInfo } from '../../app/service'
+import { AppMetaInfo } from '../../app/types'
+import { BreadCrumb, ErrorBoundary, Progressing, showError, trackByGAEvent, useBreadcrumb } from '../../common'
+import PageHeader from '../../common/header/PageHeader'
+import { ReactComponent as Settings } from '../../../assets/icons/ic-settings.svg'
+import './JobDetails.scss'
+import JobSelector from '../JobSelector/JobSelector'
+
+export default function JobDetails() {
+    const { path } = useRouteMatch()
+    const { appId } = useParams<{ appId: string }>()
+    const [jobName, setJobName] = useState('')
+    const [appMetaInfo, setAppMetaInfo] = useState<AppMetaInfo>()
+
+    useEffect(() => {
+        getAppMetaInfoRes()
```

Comment:
we should add a check for appId to be not null

Reply:
Without appId page will not open so not required. Is there any known case?

Reply:
nope